openapi3/refs.go

```diff
@@ -3,10 +3,31 @@ package openapi3
 import (
 	"context"
 	"fmt"
+	"strings"
 
 	"github.com/getkin/kin-openapi/jsoninfo"
 )
 
+type RefOrValue interface {
```

Comment:
Does this interface need all these functions?

I feel like this is actually just:
```
type MaybeRef interface {
  GetRef
  IsRef
  Resolved()
  ResetRef()
}
```

Reply:
they are used by different parts of the code, I would say yes

Reply:
I don't want to belabor this too much, but what I was commenting on is that the new RefOrValue interface isn't using `IsValid`, `IsValue`, etc. So I'm curious why they are included for the interface doesn't need a 1:1 relationship to the structures, the structures only need satisfy what the interface describes. We should be able to pair down RefOrValue to only what is required by the new functionality and not everything that is an `struct ExampleRef`, etc.

We can always add more functions later or compose a MaybeValue and MaybeRef interface into a RefOrValue:
```
type MaybeRef interface {
  IsRef()
}
type MaybeValue interface {
  IsValue()
}
type RefOrValue interface {
  MaybeRef
  MaybeValue
}
```

Reply:
you're right - these are remnants from previous versions of the code - I think there were references to Empty in oapi-codegen and also IsValid - I'll remove these. 
For now, the ones that are being used were moved to the utils file (calling site)